action/protocol/staking/protocol.go

```diff
@@ -64,84 +64,55 @@ type (
 		ReceiptStatus() uint64
 	}
 
+	// BootstrapCandidate is the candidate data need to be provided to bootstrap candidate.
+	BootstrapCandidate struct {
+		OwnerAddress      address.Address
+		OperatorAddress   address.Address
+		RewardAddress     address.Address
+		Name              string
+		SelfStakingAmount *big.Int
+	}
+
 	// Protocol defines the protocol of handling staking
 	Protocol struct {
 		addr               address.Address
-		depositGas         DepositGas
-		config             Configuration
```

Comment:
are we expecting the config to dynamically change (possibly on a per-block basis)?

the config are constants that would affect the staking behavior, any change to the config would be a hard-fork, and hence any such change will only take effect on next fork height

so i don't see the benefits of passing in the config thru ctx on a per-block basis, the current way of one-time read at NewProtocol() is both proper and good enough

Reply:
and same reasoning goes for hu